src/App.js

```diff
@@ -15,19 +15,47 @@ import CurrencyExchangeIcon from "@mui/icons-material/CurrencyExchange";
 
 function App() {
   const [expenses, setExpenses] = useState([]);
+  const [filteredExpenses, setFilteredExpenses] = useState([]);
+  const [year, setYear] = useState(-1);
+  const [month, setMonth] = useState(-1);
 
-  /**
-   * init data
-   */
+  // get data from local storage and init the states
   useEffect(() => {
     (async () => {
-      let response = await Storage.getItems();
+      const response = await Storage.getItems();
       setExpenses(response);
+      setFilteredExpenses(response);
     })();
   }, []);
 
-  const updateExpenseHandler = (expenses) => {
-    setExpenses(expenses);
+  const updateExpenseHandler = async (expense) => {
+    setExpenses((expenses) => [expense, ...expenses]);
+
+    filteredExpensesHandler(year, month, expense);
+    await Storage.addItem(expense);
+  };
+
+  // updates the state of filtered data ("-1" means to show all months/years)
+  const filteredExpensesHandler = (year, month, expense = null) => {
+    setYear(year);
+    setMonth(month);
+
+    let filteredData = expenses;
+    if (expense) {
+      filteredData = [expense, ...expenses];
+    }
+
+    if (year !== -1) {
+      filteredData = filteredData.filter(
+        (d) => new Date(d.date).getFullYear() === year
+      );
+    }
+    if (month !== -1) {
+      filteredData = filteredData.filter(
+        (d) => new Date(d.date).getMonth() === month
+      );
+    }
+    setFilteredExpenses(filteredData);
```

Comment:
`year` and `month` are the same names in useState,
change or send ``setStateAction`` to props\onChange